extensions/json-language-features/server/src/jsonServerMain.ts

```diff
@@ -253,6 +270,37 @@ function updateConfiguration() {
 			}
 		});
 	}
+
+	if (schemaStoreEnabled && schemaStoreSchemas && schemaStoreSchemas.length !== 0) {
+		schemaStoreSchemas.forEach(schema => {
+			let uri = schema.url;
+			let fileMatchAlreadyExist = true;
+			if (uri && schema.fileMatch && schema.fileMatch.length === 1) {
+				
+				let schemaFileMatch = schema.fileMatch[0];
+				
+				let isDisabledFileMatch = disabledFileMatches.some(
+					fileMatch => fileMatch === schemaFileMatch
+				);
+
+				fileMatchAlreadyExist = languageSettings.schemas.some(
+					settingSchema => {
+						if (settingSchema.fileMatch) {
+							return settingSchema.fileMatch.some(
```

Comment:
I agree with you but I don't know how to do a great comparison between file patterns. 
If I understand correctly, I should add informations on schema from schemastore catalog and provide a patch to [vscode-json-languageservice](https://github.com/Microsoft/vscode-json-languageservice). 

Which kind of information should be added? A flag like this? `{ uri, fileMatch: schema.fileMatch, schema: schema.schema, fromSchemaStore: true }`

Reply:
Yes,. I suggest have `LanguageSettings.schemas` to also tage `SchemaConfiguration[][]`.

A list of schemas or a list of groups of schema.

Reply:
Hello @aeschli
Sorry for not answering before to your message. Your proposal is to change `LanguageSettings.schemas`. I'm ok with that but I'm not sure if I understand correctly your proposal. 

IIUC, `LanguageSettings.schemas` will become a list of groups of schema like this: 

```
{
  "userSchemas": [], 
  "schemaStoreSchemas": []
}
```
Then in vscode-language-service, we will register all schemas in two different attributes of `JSONSchemaService`. For example `registeredSchemasIds` and `registeredSchemaStoreSchemasIds`.
Then we will update `getSchemaForResource` method in order to get schema from  `registeredSchemasIds` and if there is no result from `registeredSchemaStoreSchemasIds`.

Am I right? Is it overkill for the need?

Reply:
Yes, my proposal was very similar, just more general. Instead of a hierarchy of 2 ('userSchemas' and 'schemaStoreSchemas'), I suggest any number of groups.
Currently `LanguageSettings.schemas` is of type SchemaConfiguration[].
I would change it to
`LanguageSettings.schemas: SchemaConfiguration[] | SchemaConfiguration[][]`
If array of arrays ('array of groups') is passed we look at the first group (SchemaConfiguration[0]), if nothing found at the next group (SchemaConfiguration[1]) and so on.
I have another use case for that.

Reply:
I like your proposal. I will change the type to `LanguageSettings.schemas: SchemaConfiguration[] | SchemaConfiguration[][]` and update all related code in json-language-feature and vscode-json-languageservice. 
Can you explain what is your other use case?

Reply:
Our settings form a hierarchy too. You can have folder settings, workspace settings, user settings.
On each level you can define schema associations. Right now we do some merging which is not really correct. That could be fixed by the new API.